src/hooks/useBalance.ts

```diff
@@ -41,13 +39,13 @@ const initValues = {
 };
 
 const useBalance = (api: ApiPromise, address: string, providedSi: boolean = false): State => {
-  const { dispatchMessage } = useUpdateMessageContext();
   const [state, setState] = useMountedState<State>(initValues);
 
   useEffect((): (() => void) => {
-    const getBalance = async (api: ApiPromise, address: string, setState: any): Promise<VoidFn> => {
+    let unsubscribe: Promise<VoidFn>;
```

Comment:
Can we use `useApiSubscription` here?

Reply:
of course ... let me open a new PR only for this so we don't get keep making this PR bigger.